### ISSUE TITLE
complex2real dispatches wrong canonicalizer for ``Zero`` cone constraints.

### ISSUE THREAD

**Issue by rileyjmurray** [cvxpy/cvxpy#1910]
Consider the following trivial test case:
```
import numpy as np
import cvxpy as cp
from cvxpy.constraints.zero import Equality, Zero

def make_problem(use_zero: bool):
   # min{ real(x) s.t. x \in C, x == 0, real(x) >= -1, imag(x) <= 1.0 }
    x = cp.Variable(complex=True)
    constraints = [cp.real(x) >= -1.0, cp.imag(x) <= 1.0]
    if use_zero:
        constraints.append(Zero(x))
    else:
        constraints.append(Equality(x, cp.Constant(np.array(0.0)))
    prob = cp.Problem(cp.Minimize(cp.real(x)), constraints)
    return prob
``` 
Running ``make_prob(True).solve(verbose=True)`` raises an error, while running ``make_prob(False).solve(verbose=True)`` succeeds. The issue is in these lines:
https://github.com/cvxpy/cvxpy/blob/cb2170114b267be366beccf6db97af264f042ee9/cvxpy/reductions/complex2real/atom_canonicalizers/__init__.py#L96-L97
Specifically, notice that ``Zero`` maps to ``equality_canon`` when it should map to ``zero_canon``.

**Comment by rileyjmurray** [comment]
For posterity: the reason why we found this is because @aryamanjeendgar was extending the ``OpRelConeQuad`` canonicalizer to work with Hermitian inputs. We were using ``Zero(expr)`` constraints in the canonicalizer where ``expr`` was complex-valued. At first we reformulated things to work with ``Equality(expr, zero_expr)``, but then I noticed that ``equality_canon`` was being called on ``Zero`` constraint objects.
